build-release-tools/lib/PlatformClients.py

```diff
@@ -209,3 +256,93 @@ def upload_a_file(self, package, version, file_path):
         except Exception, ex:
             raise RuntimeError("Failed to upload file {0} due to {1}".format(file_path, ex))
         return True
+
+    def get_package_versions(self, package):
+        """
+        Get the version list of one package.
+        """
+        get_package_versions_url = "/".join([self._api_url, "packages", self._subject, self._repo, package])
+        versions = []
+        resp = self.session.get(get_package_versions_url)
+        if resp.ok:
+            versions = resp.json()["versions"]
+        elif resp.status_code == 404:
+            print "Bintray package: {0}/{1}/{2} doesn't exist".format(self._subject, self._repo, package)
+        else:
+            raise Exception("Failed to get package versions: {0}\n{1}".format(package, resp.text))
+        return versions
+
+    def del_package_version(self, package, version):
+        """
+        Delete a version of one package
+        """
+        del_package_version_url = "/".join([self._api_url, "packages", self._subject, self._repo, package, "versions", version])
+        print del_package_version_url
+        resp = self.session.delete(del_package_version_url)
+        if resp.ok:
+            print "Delete {0}/{1} successfully!".format(package, version)
+        else:
+            raise Exception("Failed to delete {0}/{1}\n {2}!".format(package, version, resp.text))
+
+class Dockerhub(object):
+    """
+    A client class of Dockerhub.
+    """
+    def __init__(self, creds, repo, api_url):
+        assert creds, "creds is None!"
+        assert repo, "repo is None!"
+
+        try:
+            # get from env var
+            self.username, self.password = common.parse_credential_variable(creds)
+        except ValueError as e:
+            # not in env var, parse as user:pass
+            self.username, self.password = creds.split(":")
+        self.repo = repo
+        self.api_url = api_url or "https://hub.docker.com/v2"
+
+        self.token = self.retrieve_token()
+        self.session = requests.Session()
+        self.session.headers.update({'authorization': "JWT %s" % self.token})
+
+    def retrieve_token(self):
+        """
+        Get access token of dockerhub
+        """
+        login_url = "/".join([self.api_url, "users", "login"])
+        data = {"username": self.username, "password": self.password}
+        resp = requests.post(login_url, data=data)
+        if resp.ok:
+            return resp.json()["token"]
+        else:
+            raise Exception("Failed to retrive dockerhub token.\n{0}".format(resp.text))
+
+    def get_package_tags(self, package):
+        """
+        Get all tags of a package
+        """
+        assert package, "arg package is None!"
+        get_package_versions_url = "/".join([self.api_url, "repositories", self.repo, package, "tags"])
+        resp = self.session.get(get_package_versions_url)
+        tags = []
+        if resp.ok:
+            for tag in resp.json()["results"]:
```

Comment:
I'm not sure "one docker instance" means "one image" or "one package"?
In dockerhub, repo means "rackhd", package means "on-xxxx".